sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -578,27 +573,35 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         AppendData.byName(v2Relation, df.logicalPlan, extraOptions.toMap)
 
       case (SaveMode.Overwrite, _) =>
-        ReplaceTableAsSelect(
-          catalog,
-          ident,
-          partitioningAsV2,
+        ReplaceTableAsSelectStatement(
+          nameParts,
           df.queryExecution.analyzed,
-          Map(TableCatalog.PROP_PROVIDER -> source) ++ getLocationIfExists,
+          partitioningAsV2,
+          None,
+          Map.empty,
+          Some(source),
+          Map.empty,
+          extraOptions.get("path"),
```

Comment:
Should this use the constants in `TableCatalog`?

Reply:
the table location constant is "location", which is different from the "path" in `DataFrameReader/Writer`. I'll use the constant for the comment.